src/Core/Animation/Skeleton.hpp

```diff
@@ -41,11 +41,13 @@ class RA_CORE_API Skeleton : public Handle
      * @param MODE   SpaceType of T (either SpaceType::LOCAL or SpaceType::GLOBAL)
      * @param label  the name for the new joint
      * @return       the index of the new joint
+     * \note In order to insert the root node, use \code uint( -1 ) \endcode as
+     *       its parent's index.
      */
-    int addBone( const int parent     = -1,
-                 const Transform& T   = Transform::Identity(),
-                 const SpaceType MODE = SpaceType::LOCAL,
-                 const Label label    = "" );
+    uint addBone( const uint parent    = uint( -1 ),
```

Comment:
-1 is not an uint. Forcing it to be an uint is machine dependent ... not a good idea to rely on machine dependent behavior.